### PR TITLE
ci(js): gate expensive e2e matrices and builds on unit test pass

### PR DESCRIPTION
**Pull request by mcous**
## Overview

After the huge CI runner backlog today caused by a performance degradation in Apple's notarization service, I have a few proposals for reducing our CI load. This PR is one of them!

This PR gates more expensive tests in the front-end CI workflows behind passing the initial unit test job. This has a couple downsides:

1. Unit tests must pass for sandbox assets to be built and published. If you're working out of sandbox builds, you must ensure the unit test suite is passing before pushing
2. These workflows will be a little less parallel, and may take several minutes longer (i.e. the length of time the unit tests take to run)

For (1), I'd posit that we're ignoring the vast majority of these builds, anyway, especially if the unit tests aren't passing. If you _are_ relying on the ability of sandbox builds to build with a failing test suite, you... shouldn't be doing that.

For (2), I don't think our workflow time is the thing that's getting us, it's machine usage. This strategy will tie up fewer machines.

Overall, I think the upside for all developers (not just FE devs) in fewer CI machines getting tied up greatly outweighs the downsides of this approach.

## Changelog

- Gate expensive tests and builds behind the passing of unit tests
    - App-shell OS-matrix tests + app OS-matrix builds
    - Component library storybook builds
    - Labware library OS-matrix E2E tests
    - Labware library builds
- Add 30 minute timeout to app build step
    - Will fail the job if notarization takes too long
    - Today I found a job waiting more then 5 hours for notarization
    - Default job timeout in GH Actions is 6 hours
- Remove duplicate Labware Library E2E workflow
    - The same exact E2E matrix was already present in the regular LL workflow 
- ~Remove some redundant `branch: "*"` filters that weren't doing anything~ Update branch filters to `**` to accept branches including slashes

## Review requests

👍 if you think these changes make sense

## Risk assessment

Low. Increasing the strictness of our CI jobs

### CODE REVIEW

**Comment by sfoster1** [comment]
@Opentrons/app-and-uis  @mcous Do you think it makes sense to make sandbox builds and full app builds actually opt-in with comment handler workflows? How many FE prs do we actually use the CI artifacts from vs local builds?

**Comment by mcous** [comment]
It's an interesting question. I think the main benefit of the sandbox builds isn't the sandbox publish, but the fact that CI is veryfying the change set not only passes tests, but doesn't break the build. The publish to sandbox is almost incidental, but also the lightest-weight part of the check

**Comment by SyntaxColoring** [comment]
> Remove some redundant branch: "*" filters that weren't doing anything

I don't know about these workflows specifically, but those filters might have actually been doing something despite looking redundant. 

See this note in another workflow:

https://github.com/Opentrons/opentrons/blob/1f34210d1b53ff4ffbed1830139edeff2bcce599/.github/workflows/docs-build.yaml#L17-L22

Added in [this PR](https://github.com/opentrons/opentrons/pull/8453).

Also note the `**` wildcard instead of `*`. I haven't tested this, but it might be necessary to handle branch names with a slash in them, like `feat/foo`. See [the wildcard docs](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet).

**Review by b-cooper — APPROVED** [review]
These changes all sound reasonable to me ⚖️

File: .github/workflows/app-test-build-deploy.yaml [hunk 1]
```diff
@@ -131,6 +132,7 @@ jobs:
       # build the app and deploy it
       - if: github.event_name != 'pull_request'
         name: 'build app for ${{ matrix.os }}'
+        timeout-minutes: 30
```
Comment (mcous) [inline .github/workflows/app-test-build-deploy.yaml]: 30 minutes may be too low based on preliminary results from #9949

```suggestion
        timeout-minutes: 60
```

**Pull request merged**